### ISSUE TITLE
user_ldap app: blank page if no "displayname" attribute is given

### ISSUE THREAD

**Issue by xuhdev** [owncloud/core#22541]
### Steps to reproduce
Set up user_ldap. Also, set up an LDAP server such that uid, mail, cn, sn and userPassword are all given for each user, but make sure no displayname is given in the LDAP database. Now try to log in with the correct/incorrect password with a user.

### Expected behaviour
Log in successfully with the correct password, or log in fail with the incorrect password.

### Actual behaviour
The page turns blank and the log in fails.

### Server configuration
**Operating system**: GNU/Linux

**Web server:** Apache

**Database:** MariaDB 10.1

**PHP version:** 5.6

**ownCloud version:** 8.2.2

**Updated from an older ownCloud or fresh install:**
fresh install

**The content of config/config.php:**

```
'ldapIgnoreNamingRules' => false,
```

**Are you using external storage? no
**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** LDAP

#### LDAP configuration (delete this part if not used)

```
+-------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------
| Configuration                 | s01                                                                                                                                       |
+-------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------+
| hasMemberOfFilterSupport      | 1                                                                                                                                         |
| hasPagedResultSupport         |                                                                                                                                           |
| homeFolderNamingRule          |                                                                                                                                           |
| lastJpegPhotoLookup           | 0                                                                                                                                         |
| ldapAgentName                 |                                                                                                                                           |
| ldapAgentPassword             | ***                                                                                                                                       |
| ldapAttributesForGroupSearch  |                                                                                                                                           |
| ldapAttributesForUserSearch   |                                                                                                                                           |
| ldapBackupHost                |                                                                                                                                           |
| ldapBackupPort                |                                                                                                                                           |
| ldapBase                      | dc=example,dc=com                                                                                                                             |
| ldapBaseGroups                | dc=example,dc=com                                                                                                                             |
| ldapBaseUsers                 | dc=example,dc=com                                                                                                                             |
| ldapCacheTTL                  | 600                                                                                                                                       |
| ldapConfigurationActive       | 1                                                                                                                                         |
| ldapEmailAttribute            | mail                                                                                                                                      |
| ldapExperiencedAdmin          | 0                                                                                                                                         |
| ldapExpertUUIDGroupAttr       |                                                                                                                                           |
| ldapExpertUUIDUserAttr        |                                                                                                                                           |
| ldapExpertUsernameAttr        |                                                                                                                                           |
| ldapGroupDisplayName          | cn                                                                                                                                        |
| ldapGroupFilter               |                                                                                                                                           |
| ldapGroupFilterGroups         |                                                                                                                                           |
| ldapGroupFilterMode           | 0                                                                                                                                         |
| ldapGroupFilterObjectclass    |                                                                                                                                           |
| ldapGroupMemberAssocAttr      | uniqueMember                                                                                                                              |
| ldapHost                      | ldap                                                                                                                                      |
| ldapIgnoreNamingRules         |                                                                                                                                           |
| ldapLoginFilter               | (&(&(|(objectclass=inetOrgPerson))(|(memberof=cn=owncloud,ou=groups,dc=example,dc=com)))(|(uid=%uid)(|(mailPrimaryAddress=%uid)(mail=%uid)))) |
| ldapLoginFilterAttributes     |                                                                                                                                           |
| ldapLoginFilterEmail          | 1                                                                                                                                         |
| ldapLoginFilterMode           | 0                                                                                                                                         |
| ldapLoginFilterUsername       | 1                                                                                                                                         |
| ldapNestedGroups              | 0                                                                                                                                         |
| ldapOverrideMainServer        |                                                                                                                                           |
| ldapPagingSize                | 500                                                                                                                                       |
| ldapPort                      | 389                                                                                                                                       |
| ldapQuotaAttribute            |                                                                                                                                           |
| ldapQuotaDefault              |                                                                                                                                           |
| ldapTLS                       | 0                                                                                                                                         |
| ldapUserDisplayName           | displayName                                                                                                                               |
| ldapUserFilter                | (&(|(objectclass=inetOrgPerson))(|(memberof=cn=owncloud,ou=groups,dc=example,dc=com)))                                                        |
| ldapUserFilterGroups          | owncloud                                                                                                                                  |
| ldapUserFilterMode            | 0                                                                                                                                         |
| ldapUserFilterObjectclass     | inetOrgPerson                                                                                                                             |
| ldapUuidGroupAttribute        | auto                                                                                                                                      |
| ldapUuidUserAttribute         | auto                                                                                                                                      |
| turnOffCertCheck              | 0                                                                                                                                         |
| useMemberOfToDetectMembership | 1                                                                                                                                         |
+-------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------+

```

### Client configuration
**Browser:** Firefox

**Operating system:** GNU/Linux

#### ownCloud log (data/owncloud.log)
```
{"reqId":"MOADCvda48MXBs53t9kt","remoteAddr":"172.17.0.20","app":"user_ldap","message":"initializing paged search for  Filter objectClass=* base Array\n(\n    [0] => uid=hong,ou=people,dc=example,dc=com\n)\n attr Array\n(\n    [0] => entryuuid\n)\n limit 500 offset 0","level":0,"time":"2016-02-21T10:58:19+00:00","method":"POST","url":"\/"}
{"reqId":"MOADCvda48MXBs53t9kt","remoteAddr":"172.17.0.20","app":"user_ldap","message":"Ready for a paged search","level":0,"time":"2016-02-21T10:58:19+00:00","method":"POST","url":"\/"}
{"reqId":"MOADCvda48MXBs53t9kt","remoteAddr":"172.17.0.20","app":"user_ldap","message":"initializing paged search for  Filter objectClass=* base Array\n(\n    [0] => uid=hong,ou=people,dc=example,dc=com\n)\n attr Array\n(\n    [0] => displayName\n)\n limit 500 offset 0","level":0,"time":"2016-02-21T10:58:19+00:00","method":"POST","url":"\/"}
{"reqId":"MOADCvda48MXBs53t9kt","remoteAddr":"172.17.0.20","app":"user_ldap","message":"Ready for a paged search","level":0,"time":"2016-02-21T10:58:19+00:00","method":"POST","url":"\/"}
{"reqId":"MOADCvda48MXBs53t9kt","remoteAddr":"172.17.0.20","app":"user_ldap","message":"Requested attribute displayname not found for uid=hong,ou=people,dc=example,dc=com","level":0,"time":"2016-02-21T10:58:19+00:00","method":"POST","url":"\/"}
{"reqId":"MOADCvda48MXBs53t9kt","remoteAddr":"172.17.0.20","app":"user_ldap","message":"No or empty name for uid=hong,ou=people,dc=example,dc=com.","level":1,"time":"2016-02-21T10:58:19+00:00","method":"POST","url":"\/"}
{"reqId":"MOADCvda48MXBs53t9kt","remoteAddr":"172.17.0.20","app":"PHP","message":"Call to a member function processAttributes() on null at \/var\/www\/html\/apps\/user_ldap\/lib\/access.php#717","level":3,"time":"2016-02-21T10:58:19+00:00","method":"POST","url":"\/"}

```

**Comment by xuhdev** [comment]
Even the display name is not given, it's probably even better for server to always explicitly fail the login, or fall back the display name to something else, instead of giving a blank page.

**Comment by blizzz** [comment]
Fixed with https://github.com/owncloud/core/pull/20804 for upcoming 8.2.3

**Comment by it25fg** [comment]
Seems the fix was only merged into 9.0.x branch. 8.2.4 is out now, and even there the patch did not show up. Did I miss something?

**Comment by blizzz** [comment]
@it25fg you are totally right. The backport did not happen and the label information in #20804 (as of now) is wrong. I do not really know why and how this mistake happened, but will do the backport now. Thank you  for notifying us!
